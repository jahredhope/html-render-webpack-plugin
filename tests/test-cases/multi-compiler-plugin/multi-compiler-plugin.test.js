const MemoryFS = require("memory-fs");
const webpack = require("webpack");
const path = require("path");

const config = require("./webpack.config");
const HtmlRenderPlugin = require("../../../src");
const getDirContentsSync = require("../../utils/getDirContentsSync");

describe("Plugin applied to MultiCompiler", () => {
  const renderDirectory = path.join(process.cwd(), "dist", "render");

  it("should render a HTML file", async done => {
    const compiler = webpack(config);

    const memoryFs = new MemoryFS();
    compiler.outputFileSystem = memoryFs;

    new HtmlRenderPlugin().apply(compiler);

    compiler.run(error => {
      expect(error).toBe(null);
      const contents = memoryFs.readFileSync(
        path.join(process.cwd(), "dist", "index.html"),
        "utf8"
      );
      expect(contents).toMatchSnapshot();
      done();
    });
  });
  it("should render in a custom directory", async done => {
    const compiler = webpack(config);

    const memoryFs = new MemoryFS();
    compiler.outputFileSystem = memoryFs;

    new HtmlRenderPlugin({ renderDirectory }).apply(compiler);

    compiler.run(error => {
      expect(error).toBe(null);
      const contents = getDirContentsSync(renderDirectory, {
        fs: memoryFs
      });
      expect(contents).toMatchSnapshot();
      done();
    });
  });
  it("should render a file per route", async done => {
    const compiler = webpack(config);

    const memoryFs = new MemoryFS();
    compiler.outputFileSystem = memoryFs;

    new HtmlRenderPlugin({
      routes: ["", "pageA", "pageB", "error.html"],
      renderDirectory
    }).apply(compiler);

    compiler.run(error => {
      expect(error).toBe(null);
      const contents = getDirContentsSync(renderDirectory, { fs: memoryFs });
      expect(contents).toMatchSnapshot();
      done();
    });
  });

  it("should render routes with extra information", async done => {
    const compiler = webpack(config);

    const memoryFs = new MemoryFS();
    compiler.outputFileSystem = memoryFs;

    new HtmlRenderPlugin({
      routes: [
        { route: "production", language: "en-us", environment: "production" },
        {
          route: "development",
          language: "en-us",
          environment: "development"
        }
      ],
      renderDirectory
    }).apply(compiler);

    compiler.run(error => {
      expect(error).toBe(null);
      const contents = getDirContentsSync(renderDirectory, { fs: memoryFs });
      expect(contents).toMatchSnapshot();
      done();
    });
  });
  it("should allow custom file paths", async done => {
    const compiler = webpack(config);

    const memoryFs = new MemoryFS();
    compiler.outputFileSystem = memoryFs;

    new HtmlRenderPlugin({
      transformFilePath: ({ route, language, environment }) =>
        `/${environment}/${language}/${route}`,
      routes: [
        { route: "about/us", language: "en-us", environment: "production" },
        {
          route: "about/us",
          language: "en-au",
          environment: "development"
        }
      ],
      renderDirectory
    }).apply(compiler);

    compiler.run(error => {
      expect(error).toBe(null);
      expect(
        memoryFs.existsSync(
          path.join(renderDirectory, "/production/en-us/about/us/index.html")
        )
      ).toBe(true);
      expect(
        memoryFs.existsSync(
          path.join(renderDirectory, "/development/en-au/about/us/index.html")
        )
      ).toBe(true);
      expect(
        memoryFs.existsSync(path.join(renderDirectory, "/about/us/index.html"))
      ).toBe(false);
      const contents = getDirContentsSync(renderDirectory, { fs: memoryFs });
      expect(contents).toMatchSnapshot();

      done();
    });
  });
});
