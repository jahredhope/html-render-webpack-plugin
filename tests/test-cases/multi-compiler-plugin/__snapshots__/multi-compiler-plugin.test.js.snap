// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin applied to MultiCompiler should allow custom file paths 1`] = `
Object {
  "development": Object {
    "en-au": Object {
      "about": Object {
        "us": Object {
          "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"about/us\\",
      \\"language\\": \\"en-au\\",
      \\"environment\\": \\"development\\"
    }
  ]</code>
</body>
</html>",
        },
      },
    },
  },
  "production": Object {
    "en-us": Object {
      "about": Object {
        "us": Object {
          "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"about/us\\",
      \\"language\\": \\"en-us\\",
      \\"environment\\": \\"production\\"
    }
  ]</code>
</body>
</html>",
        },
      },
    },
  },
}
`;

exports[`Plugin applied to MultiCompiler should render a HTML file 1`] = `
"<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"\\"
    }
  ]</code>
</body>
</html>"
`;

exports[`Plugin applied to MultiCompiler should render a file per route 1`] = `
Object {
  "error.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"error.html\\"
    }
  ]</code>
</body>
</html>",
  "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"\\"
    }
  ]</code>
</body>
</html>",
  "pageA": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"pageA\\"
    }
  ]</code>
</body>
</html>",
  },
  "pageB": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"pageB\\"
    }
  ]</code>
</body>
</html>",
  },
}
`;

exports[`Plugin applied to MultiCompiler should render in a custom directory 1`] = `
Object {
  "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"\\"
    }
  ]</code>
</body>
</html>",
}
`;

exports[`Plugin applied to MultiCompiler should render routes with extra information 1`] = `
Object {
  "development": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"development\\",
      \\"language\\": \\"en-us\\",
      \\"environment\\": \\"development\\"
    }
  ]</code>
</body>
</html>",
  },
  "production": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"production\\",
      \\"language\\": \\"en-us\\",
      \\"environment\\": \\"production\\"
    }
  ]</code>
</body>
</html>",
  },
}
`;
